web/src/elements/wizard/Wizard.ts

```diff
@@ -20,91 +21,149 @@ export const ApplyActionsSlot = "apply-actions";
+            `,
+        );
+    }
+
+    //#region Properties
+
+    /**
+     * Whether the wizard can be cancelled.
+     */
     @property({ type: Boolean })
     canCancel = true;
 
+    /**
+     * Whether the wizard can go back to the previous step.
+     */
     @property({ type: Boolean })
     canBack = true;
 
+    /**
+     * Header title of the wizard.
+     */
     @property()
     header?: string;
 
+    /**
+     * Description of the wizard.
+     */
     @property()
     description?: string;
 
+    /**
+     * Whether the wizard is valid and can proceed to the next step.
+     */
     @property({ type: Boolean })
     isValid = false;
 
-    static get styles(): CSSResult[] {
-        return super.styles.concat(
-            PFWizard,
-            css`
-                .pf-c-modal-box {
-                    height: 75%;
-                }
-            `,
-        );
-    }
+    /**
+     * Actions to display at the end of the wizard.
+     */
+    @property({ attribute: false })
+    actions: WizardAction[] = [];
 
+    @property({ attribute: false })
+    finalHandler = () => {
+        return Promise.resolve();
+    };
+
+    @property({ attribute: false })
+    state: { [key: string]: unknown } = {};
+
+    //#endregion
+
+    //#region State
+
+    /**
+     * Memoized step tag names.
+     */
     @state()
     _steps: string[] = [];
 
+    /**
+     * Step tag names present in the wizard.
+     */
     get steps(): string[] {
         return this._steps;
     }
 
-    set steps(steps: string[]) {
-        const addApplyActionsSlot = this.steps.includes(ApplyActionsSlot);
-        this._steps = steps;
+    set steps(nextSteps: string[]) {
+        const addApplyActionsSlot = this._steps.includes(ApplyActionsSlot);
+
+        this._steps = nextSteps;
+
         if (addApplyActionsSlot) {
             this.steps.push(ApplyActionsSlot);
         }
-        this.steps.forEach((step) => {
-            const exists = this.querySelector(`[slot=${step}]`) !== null;
-            if (!exists) {
-                const el = document.createElement(step);
-                el.slot = step;
-                el.dataset["wizardmanaged"] = "true";
-                this.appendChild(el);
-            }
-        });
+
+        for (const step of this._steps) {
+            const existingStepElement = this.getStepElementByName(step);
+
+            if (existingStepElement) continue;
+
+            const stepElement = document.createElement(step);
+
+            stepElement.slot = step;
+            stepElement.dataset["wizardmanaged"] = "true";
+
+            this.appendChild(stepElement);
+        }
+
         this.requestUpdate();
     }
 
+    /**
+     * Initial steps to reset to.
+     */
     _initialSteps: string[] = [];
 
-    @property({ attribute: false })
-    actions: WizardAction[] = [];
-
     @state()
-    _currentStep?: WizardPage;
+    _activeStep?: WizardPage;
 
-    set currentStep(value: WizardPage | undefined) {
```

Comment:
Preferred use of `active` over `current` to emphasize the state of the step, rather than its order in the array. Also to differentiate between `ref.current`.